core/src/test/java/edu/wpi/grip/core/operations/networktables/NTPublishOperationTest.java

```diff
@@ -0,0 +1,38 @@
+package edu.wpi.grip.core.operations.networktables;
+
+import org.junit.Test;
+
+/**
+ * Tests for error handling in the publish operations' reflection stuff.
+ */
+public class NTPublishOperationTest {
+
+    private class ReportWithNonDistinctWeights implements NTPublishable {
+        @NTValue(key = "foo", weight = 1)
+        public double getFoo() {
+            return 0.0;
+        }
+
+        @NTValue(key = "bar", weight = 1)
+        public double getBar() {
+            return 0.0;
+        }
+    }
+
+    private class ReportWithParameters implements NTPublishable {
+        @NTValue(key = "foo", weight = 0)
+        public double getFoo(boolean b) {
+            return b ? 1.0 : 0.0;
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNonDistinctWeights() {
+        new NTPublishOperation<>(ReportWithNonDistinctWeights.class);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidParameters() {
+        new NTPublishOperation<>(ReportWithParameters.class);
+    }
```

Comment:
Add a test that `getInputSockets` returned the expected number of sockets in the expected order.